### PR TITLE
Add comments and escapes to ECR docs

### PR DESCRIPTION
**Pull request by KCErb**
PR #2004 added some new features to ECR, here are the docs to go with them.

Also, I felt like the language was a little awkward in the examples. Originally the class: `Greeting` would greet a user with the phrase `Greeting, KC` but I believe it is more natural to call the class a `Greeter` and I'm sure (as a native English speaker) that the idiom is "Greetings, KC".

Sorry if that kind of input isn't welcome, I don't mean to be pedantic. I'm happy to revert that aspect of this PR if requested :)

### CODE REVIEW

File: src/ecr.cr [hunk 1]
```diff
@@ -12,61 +12,98 @@
...
 #
-#   ECR.def_to_s "greeting.ecr"
+#   ECR.def_to_s "greeter.ecr"
 # end
 #
-# # greeting.ecr
-# Greeting, <%= @name %>!
-#
-# Greeting.new("John").to_s #=> Greeting, John!
+# Greeter.new("John").to_s # => Greetings, John!
```
Comment (straight-shoota) [inline src/ecr.cr]: Could you wrap the result in quotes?

File: src/ecr.cr [hunk 2]
```diff
@@ -12,61 +12,98 @@
...
+#
+# class Greeter
+#   def initialize(@name : String)
+#   end
+#
+#   ECR.def_to_s "greeter.ecr"
+# end
 #
-# Greeting.new(nil).to_s #=> Greeting!
+# Greeter.new(nil).to_s # => Greetings!
```
Comment (straight-shoota) [inline src/ecr.cr]: Ditto: quotes.
And maybe add a usage example with a name?

File: src/ecr.cr [hunk 3]
```diff
@@ -12,61 +12,98 @@
...
 #
-#   ECR.def_to_s "greeting.ecr"
+#   ECR.def_to_s "greeter.ecr"
 # end
 #
-# # greeting.ecr
-# <%- @names.each do |name| -%>
-# Hi, <%= name %>!
-# <%- end -%>
+# Greeter.new("John", "Zoe", "Ben").to_s
```
Comment (straight-shoota) [inline src/ecr.cr]: ` # => "Hi, John!\nHi, Zoe!\nHi, Ben!"`

File: src/ecr.cr [hunk 4]
```diff
@@ -12,61 +12,98 @@
...
+# ```
 #
-# Greeting.new("John", "Zoe", "Ben").to_s
-# #=> Hi, John!
-# #=> Hi, Zoe!
-# #=> Hi, Ben!
+# ```
+# require "ecr"
+# foo = 2
+# string = ECR.render("demo.ecr")
```
Comment (straight-shoota) [inline src/ecr.cr]: There is no need for storing the result in a variable, just directly show the result. `.inspect` is not essential, either.

**Comment by KCErb** [comment]
@straight-shoota Thanks for the comments, I believe I implemented them all correctly.

**Pull request merged**